ingestion/ingestion/storage/gcp.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Any, Iterable
+
+from attr import dataclass
+from google.cloud.storage import Blob, Bucket
+
+from ingestion.hash import Crc32cCalculator
+
+
+@dataclass
+class FakeBlob:
+    def upload_from_string(self, *args, **kwargs): ...
+
+
+@dataclass
+class FakeBucket:
+    name: str
+    lifecycle_rules: list[Any] = []
+
+    def get_blob(self, *_): ...
+    def blob(self, *_) -> Blob:
+        return FakeBlob()  # type:ignore
+
+    def patch(self, *_): ...
+    def list_blobs(self, *_, **kwargs) -> Iterable[Any]: ...
+
+
+class RemoteStorage:
+    """Abstracts uploading files to GCP blob storage."""
+
+    bucket: Bucket | FakeBucket
+    dry_run: bool
+
+    _logger: logging.Logger
+
+    def __init__(
+        self,
+        bucket: Bucket | FakeBucket,
+        *,
+        dry_run: bool = False,
+    ) -> None:
+        self.bucket = bucket
+        self.dry_run = dry_run
+
+        self._logger = logging.getLogger(f"{__name__} ({bucket.name=})")
+        if dry_run:
+            self._logger.setLevel(logging.DEBUG)
```

Comment:
This was set to show dry run logs in a muted gray (debug color format) in the terminal. They are now in info, should we remove this or change the logs to debug (cyan blue color)?

Reply:
I think it's better to remove the debug level from the gcp module. The debug level is for more fine grain information, a dry run is about information. Also, if we activate the debug level on a global level later, it would mean that those messages (the one for dry run if we let them as debug) will be mixed with a lot of other messages from other modules and it will be complicated to see the trace.